src/targets/gpu/include/migraphx/gpu/gemm.hpp

```diff
@@ -132,6 +133,43 @@ struct rocblas_gemm
     {
         return shapes.size() - 1;
     }
+
+    void finalize(context& ctx, const shape& output_shape, const std::vector<shape>& input_shapes)
+    {
+        rocblas_initialize();
+        rocblas_status_ check_valid;
+        if(ctx.get_exhaustive_tune_flag() && solution_idx == 0)
+        {
+            if(this->name() == "gpu::gemm")
+            {
+                auto gemmImpl = gemm_impl<float>(
+                    output_shape, input_shapes, alpha, beta, int8_x4_format, compute_fp32);
+                solution_idx = gemmImpl.tune(ctx, input_shapes);
+                check_valid  = gemmImpl.validate(ctx, input_shapes, solution_idx);
```

Comment:
Only need to validate if we are de-serializing. No need to validate just after tuning since it would be valid for sure.

Reply:
Can you be more specific? How do we know if we are deserializing, and where in the code?

Reply:
After deserializing `finalize()` is called for each operator. 
I mean you can place `validate()`  inside `else_if(solution_idx !=0)`. So that it only validates for the case where tuning is not requested and solution_index is non-default.

Reply:
Done.  Refactored finalize() to call either tune() or validate() but not both.